### PR TITLE
Chromothripsis v21 update

### PR DESCRIPTION
**Pull request by LauraEgolf**
#### What scientific question is your analysis addressing?
Update chromothripsis module from v19 to v21 data (v20 update included changes to CNV consensus data, which affects ShatterSeek analysis).

#### What was your approach?
- Adjusted `02-run-shatterseek-and-classify-confidence.R` to work with new format for CNV consensus data (filled in NA values with tumor ploidy). Also added a couple debugging updates.
- Updated `03-plot-chromothripsis-by-histology.Rmd` with new plots for cancer_group. Also updated all plot titles for clarity.
- Adjusted `04-plot-chromothripsis-and-breakpoint-data.html` to accommodate increased number of chromothripsis calls (code should be more flexible now).

#### What GitHub issue does your pull request address?
#1122 (see also: #1007)

#### Which areas should receive a particularly close look?
N/A

#### Is there anything that you want to discuss further?
N/A

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes - the figure going into the manuscript will probably be `plots/03-histology/chromothripsis_proportion_by_cancerGroup.pdf` but we should discuss further. I still plan to finish out the module by adding a couple scripts to visualize candidate chromothripsis regions, but I don't think this will affect any of the results I've submitted so far.

#### What types of results are included (e.g., table, figure)?
Updated all tables and figures. Added 2 new figures under `03-histology` showing breakdown by cancer_group.

#### What is your summary of the results?
The updated CNV consensus data resulted in more chromothripsis calls (specifically, the # CNVs increased when the stringency for reciprocal CNV overlap was reduced, so ShatterSeek detected more CN oscillations). Previous results showed 64/777 tumors with chromothripsis; we are now seeing 87/777 tumors. Otherwise, most of the results look very similar. (See notes from previous PRs: #1098, #1106).

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/chromothripsis/02-run-shatterseek-and-classify-confidence.R [hunk 1]
```diff
@@ -96,7 +109,6 @@ mergeCNsegments <- function(cnv_df) {
   # does not match the previous row (starting at second row)
   cnv_df$index <- 0
   index_counter <- 1
-  if (nrow(cnv_df) >=2) {
```
Comment (LauraEgolf) [inline analyses/chromothripsis/02-run-shatterseek-and-classify-confidence.R:99]: This was Krutika's edit for debugging - I moved down below (line 163 in the updated file)

File: analyses/chromothripsis/03-plot-chromothripsis-by-histology.Rmd [hunk 2]
```diff
@@ -42,53 +43,59 @@ chromoth_per_sample <- read_tsv(file.path(analysis_dir, "results", "chromothrips
...
+## For each histology (display_group), calculate proportion of tumors with at least 
+## one chromothripsis region (`prop`)
+## This will be used in first plot, and to reorder display_group in subsequent plots
 chromoth_histology_summary <- chromoth_histology %>%
   count(any_regions_logical, display_group, hex_codes) %>%
   tidyr::spread(key = any_regions_logical, value = n, fill=0) %>%
-  group_by(display_group, hex_codes) %>%
-  mutate(group_size = sum(`TRUE`, `FALSE`)) %>%
-  mutate(prop = `TRUE` / group_size) %>%
-  mutate(labels = paste0(`TRUE`, " / ", group_size))
```
Comment (LauraEgolf) [inline analyses/chromothripsis/03-plot-chromothripsis-by-histology.Rmd:64]: This section did not actually change, I just updated the variable names to contrast with the code I added.

File: analyses/chromothripsis/03-plot-chromothripsis-by-histology.Rmd [hunk 3]
```diff
@@ -42,53 +43,59 @@ chromoth_per_sample <- read_tsv(file.path(analysis_dir, "results", "chromothrips
...
-    chromoth_histology_summary$display_group,
-    chromoth_histology_summary$prop,
-    min, .desc=TRUE)
-# Alternate code (I like this better but it didn't work)
-# chromoth_histology_summary <- chromoth_histology_summary %>%
-#   mutate(display_group = forcats::fct_reorder(
-#     display_group,
-#     prop,
-#     min, .desc=TRUE))  
-
```
Comment (LauraEgolf) [inline analyses/chromothripsis/03-plot-chromothripsis-by-histology.Rmd:79]: Same here - no substantial change, just cleaned up the code.

**Comment by jharenza** [comment]
Hi @LauraEgolf ! thanks for adding this. Could you convert the figures to png instead of pdf? Thanks!

**Comment by jharenza** [comment]
@LauraEgolf can you also add a linear regression line + R and p-value to this plot: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/084c0e13ee17561587e360406dac0fa195050f40/analyses/chromothripsis/plots/04-breakpoint-data/count_chromothripsis_cnv_and_sv_breaks_scatterplot.pdf - this is great!

and legends to the # of SV or CNV vs chromothripsis regions plots - those are really nice!

**Comment by LauraEgolf** [comment]
@jharenza Sounds great, I will make those changes today or tomorrow.
> and legends to the # of SV or CNV breaks vs chromothripsis regions plots - those are really nice!

For this, do you mean a legend for the fill? The fill just corresponds to the # chromothripsis regions, which is shown on the x axis. I can still add a legend if you want though.

**Comment by jharenza** [comment]
> For this, do you mean a legend for the fill? The fill just corresponds to the # chromothripsis regions, which is shown on the x axis. I can still add a legend if you want though.

Oh yes but in that case, then you don't need a color for the fills. Thanks!

**Review by jaclyn-taroni — APPROVED** [review]
Looks good to me, thanks @LauraEgolf !

**Pull request merged**